### PR TITLE
Test/23-create-full-path-for-tracking-file

### LINKED ISSUE #23: Create full path for tracking file
Original issue by Nickelza on 2023-02-06 00:51:59+00:00

When given a file.json to store data, it'd be nice if the path to the file is created if it doesn't exist

### PR DESCRIPTION
**Pull request by 444B**
# Pull Request Template

## TL;DR
Provide a brief summary of the changes made.
- pipfile.lock updated with latest versions
- CONTRIBUTING.md updated withn better steps for setup

## Problem Solved
1. Describe the problem your changes address. 
2. Please mention the issue number if relevant
- When creating a json log file, the whole path is not created. Current behaviour unknown
- fixes https://github.com/444B/streamlit-analytics2/issues/23

## Manual Testing
- [X] You have manually tested all changes to ensure they work as intended. Replace the [ ] with a [X] if True.

**How was this tested?**
Describe the tests that you ran to verify your changes.
- Replicated in branch and edited core files, reloaded streamlit and tested by providing a path to save_to_json()

### CODE REVIEW

**Comment by 444B** [comment]
fixes #23
